Add `displayLength` prop to toasts

glenngijsberts opened the pull request:
# Description

This will make it possible to have different timeouts/durations for showing toasts. The default is 3 seconds, which is fine for most toasts but sometimes you have a toast that's a bit longer to read and then it would be nice to have a bit longer duration, without making the toast persistent. 

## How to test

- Checkout this branch
- `yarn storybook`
- Go to the toast stories
- Verify the basic one is showing for 3 seconds
- Verify the long displayLength story is showing for 5 seconds

Marruk commented:
To me it would be nicer to make presets for this use-case, leaving the timing open allows for Toasts to be shown for too little time. The "persist" property could also be changed to be one of those duration presets, with these separate properties you could set yourself a persistent toast that times out after 2 seconds (what happens then?).

(if you agree, we could go with "short" -> current value, "long" -> 5 seconds?, "persistent")

glenngijsberts commented:
> To me it would be nicer to make presets for this use-case, leaving the timing open allows for Toasts to be shown for too little time. The "persist" property could also be changed to be one of those duration presets, with these separate properties you could set yourself a persistent toast that times out after 2 seconds (what happens then?).
> 
> (if you agree, we could go with "short" -> current value, "long" -> 5 seconds?, "persistent")

@Marruk good idea 👍 will have a look!

glenngijsberts committed:
feat(toast): use displayLength property

This propert will control the amount of time we show a toast and replace the persist property.

BREAKING CHANGE: Persist property is removed; use displayLength instead.

glenngijsberts committed:
docs(Toast): Add story for long displayLength property

RobVermeer commented on src/components/Toast/index.tsx:
```diff
@@ -234,9 +240,15 @@ function Item({
+        return
       }
+
+      timer = setTimeout(cancel, 3000)
```
Nit

```suggestion
      const delay = displayLength === ToastDisplayLength.long ? 5000 : 3000

      timer = setTimeout(cancel, delay)
```

glenngijsberts replied:
Nice, fixed 👍

RobVermeer reviewed (approved):
⏲️

RobVermeer commented:
:tada: This PR is included in version 29.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@ticketswap/solar/v/29.0.0)
- [GitHub release](https://github.com/TicketSwap/solar/releases/tag/v29.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
